@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -207,8 +230,7 @@ export class TileGeometryLoader {
         // If the tile is not ready for display, or if it has become invisible while being loaded,
         // for example by moving the camera, the tile is not finished and its geometry is not
         // created. This is an optimization for fast camera movements and zooms.
-        if (this.discardNeedlessTile(tile)) {
-            this.cancel();
+        if (this.cancelNeedlessTile(tile)) {
```

Comment:
its better then before, but I still think that it would be simpler to have just one direction of command and call on tileGeometryLoader just to do what it should do. And as hopefully we can get rid of the update function, .. I think the Tile should check if its still needed before even calling on the tileGeometryLoader.

So before even calling on tileGEometryLoader.update the tile could check for:

if (!tile.isVisible || tile.disposed) {....

and cancel the tileGEometryLoading

Reply:
Have you seen that createGeomety() (executed in the async task) is also calling cancelNeedlessTile? 
I don't see it makes much sense to remove it from here but leave it there.

When we remove the whole tile.update() workflow all this code will disappear anyway.

Reply:
yes, I saw, but I think, this should not happen, as the isExpired check for the task should cancel the task before we even get there.
But, I am ok, to wait this out, until tile.update() hopefully disappears

Reply:
The whole idea is to separate the resposibilities to avoid the "rabbithole chasing" as @nzjony  mentioned, wh should let tileGeometryLoader do what its supposed to do, that is load the geometry, .. disposing the tile it belongs to, shouldnt be part of that.

Reply:
done. Moved the check to the isExpired function for the task.